diff/diff.go

```diff
@@ -19,6 +21,91 @@ import (
 	"github.com/kong/go-kong/kong"
 )
 
+type EntityState struct {
+	Name string `json:"name"`
+	Kind string `json:"kind"`
+	Body any    `json:"body"`
+}
+
+func diffObjects(obj1, obj2 interface{}) interface{} {
+	// Convert the objects to JSON strings.
+	json1, err := json.Marshal(obj1)
+	if err != nil {
+		panic(err)
```

Comment:
Panic'ing at this point would leave the user into an uncertain state: were API queries successful or not? Where in the process did this fail?

Can we try to provide some useful information to the user instead of a bare panic? (same for the other panics in this file)

Reply:
Fixed.